core/src/main/java/org/fao/geonet/api/records/attachments/Store.java

```diff
@@ -312,6 +312,18 @@ MetadataResource putResource(ServiceContext context, String metadataUuid, String
     String delResource(ServiceContext context, String metadataUuid,  MetadataResourceVisibility metadataResourceVisibility,
                        String resourceId, Boolean approved) throws Exception;
 
+    /**
+     * Copy all resources from none approved (draft working copy) to approved folder.
+     *
+     *
+     * @param context
+     * @param sourceUuid               The source metadata UUID
+     * @param targetUuid               The target metadata UUID
+     * @param metadataResourceVisibility The type of sharing policy {@link MetadataResourceVisibility}
+     *
+     */
+    void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility) throws Exception;
```

Comment:
I believe the interface needs the source approval flag and target approval flag or it will potentially be copying data from the wrong working copy when it should be copying from the approved folder.

Reply:
yeah. Right now, I saw many of the implementation is hardcoded as approve=true, not approved=fasle flag. So I hardcoded in this way as it's doing copy from source(none approved) to target(approved). It will never be a bad idea to add more parameter to have flexibility in the future if the copy goes reverse way.

Reply:
Yes. I am working on it. The flag will be added as it will get reverse way (i.e copy from approved copy to working copy, then the user edited it, do copy from working copy back to approved copy). So add such flag can make this interface re-useable for both scenarios.

Reply:
Resolved